frontend/src/components/auth/ForgotPassword.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import {
+  Text,
+  FormControl,
+  FormLabel,
+  Input,
+  Button,
+  useToast,
+} from "@chakra-ui/react";
+import { useMutation, useQuery } from "@apollo/client";
+import { RESET_PASSWORD } from "../../APIClients/mutations/AuthMutations";
+import NavigationButtons from "./TeacherSignup/NavigationButtons";
+import { ArrowBackOutlineIcon } from "../../assets/icons";
+import { GET_USERS_BY_ROLE } from "../../APIClients/queries/UserQueries";
+import { UserResponse } from "../../APIClients/types/UserClientTypes";
+import { Role } from "../../types/AuthTypes";
+
+interface ForgotPasswordProps {
+  role: Role;
+  hasError: boolean;
+  email: string;
+  step: number;
+  setEmail: React.Dispatch<React.SetStateAction<string>>;
+  setEmailError: React.Dispatch<React.SetStateAction<boolean>>;
+  setEmailNotFoundError: React.Dispatch<React.SetStateAction<boolean>>;
+  setStep: React.Dispatch<React.SetStateAction<number>>;
+}
+
+const ForgotPassword = ({
+  role,
+  hasError,
+  email,
+  step,
+  setEmail,
+  setEmailError,
+  setEmailNotFoundError,
+  setStep,
+}: ForgotPasswordProps): React.ReactElement => {
+  const toast = useToast();
+  const { data } = useQuery(GET_USERS_BY_ROLE, {
+    variables: { role },
+  });
+
+  const [resetPassword] = useMutation<{ resetPassword: boolean }>(
+    RESET_PASSWORD,
+    {
+      onCompleted() {
+        setStep(2);
+      },
+    },
+  );
+
+  const onResetPasswordClick = async () => {
+    if (!email) {
+      setEmailError(true);
+      return;
+    }
+
+    if (!data.usersByRole?.find((user: UserResponse) => user.email === email)) {
+      setEmailNotFoundError(true);
+      return;
+    }
+
+    await resetPassword({ variables: { email } });
+  };
+
+  const onResendClick = async () => {
+    await onResetPasswordClick();
+    toast({
```

Comment:
added this toast for visual signal that email was resent - can be reviewed by design

Reply:
is so cute - maybe we can keep it for now and then during testing they can tell us if they hate it :D